### PR TITLE
Added CFN execution role

### PR DESCRIPTION
**Pull request by hencrice**
This role is needed when CodePipeline deploys the application infrastructure through the CFN deployment action.

#323

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/environment/cf.yml [hunk 1]
```diff
@@ -227,9 +227,32 @@ Resources:
...
+          Action: sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: execute
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+            - 
+              Effect: Allow
+              Action: "*"
```
Comment (SoManyHs) [inline templates/environment/cf.yml]: Is this copied from somewhere? Just anticipating appsec saying this is too permissive.

/c @allisaurus

Comment (kohidave) [reply templates/environment/cf.yml]: Please use the PowerUser managed policy and the ability to create roles which look like task execution/task role.

Comment (kohidave) [reply templates/environment/cf.yml]: (also the permissive policy is going to be mitigated by allowing folks to pass in their own policy, if they want `archer env init --app-infrastructure-policy app-policy.json`)

Comment (hencrice) [reply templates/environment/cf.yml]: Nope not copied I wrote that myself based on a previous casual discuss I had with David. But 100% agree Daivid's suggestion is better.

**Comment by hencrice** [comment]
Talked to David about the problem that will occur when creating the task role if the CFN execution role is not given enough permission. We agree to make it permissive now and scope it down later.

**Pull request merged**